src/app/common/theme-provider.tsx

```diff
@@ -27,12 +27,23 @@ const ThemeContext = createContext<{
 const getSystemTheme = () =>
   window.matchMedia('prefers-color-scheme: dark').matches ? 'dark' : 'light';
 
+const localStorageUserSelectedThemeKey = 'user-selected-theme';
+function getInitialTheme() {
+  return (localStorage.getItem(localStorageUserSelectedThemeKey) ??
+    getSystemTheme()) as ComputedTheme;
+}
+
 interface ThemeSwitcherProviderProps {
   children: JSX.Element | JSX.Element[];
 }
 export const ThemeSwitcherProvider = ({ children }: ThemeSwitcherProviderProps) => {
-  const [userSelectedTheme, setUserSelectedTheme] = useState<UserSelectedTheme>('system');
-  const [theme, setTheme] = useState<ComputedTheme>(() => getSystemTheme());
+  const [userSelectedTheme, privateSetUserSelectedTheme] = useState<UserSelectedTheme>('system');
+  const [theme, setTheme] = useState<ComputedTheme>(() => getInitialTheme());
+
+  function setUserSelectedTheme(theme: UserSelectedTheme) {
+    localStorage.setItem(localStorageUserSelectedThemeKey, theme);
+    privateSetUserSelectedTheme(theme);
+  }
```

Comment:
Could you put this in Redux, such as in a new `settings` slice that's whitelisted for `redux-persist`? 

Reason for this ask is that we previously we had a lot of `localStorage` operations spread over the codebase, and did our best to remove them, and manage persisted state with a single tool. This helps us with migrations when needed, and cleaning up old state when it's no longer used.

For example, my main wallet install that's been around for ages, has much dead/unused localstorage saved
<img width="486" alt="image" src="https://user-images.githubusercontent.com/1618764/199751217-3f818e33-378b-4ac2-952d-94057e8361fc.png">

Further, perhaps we need to access this in the background script at a later point. In MV3, the background script cannot read `localStorage`. Our redux persist set up uses `chrome.storage`

Reply:
Sure, can be done. I'm not familiar with redux slices, so may take a while.

Reply:
This isn't a hugely urgent issue, so take whatever time you need. Happy to jump on a call and discuss as well

Reply:
Alright, new PR up using redux with persisted state, seems to work :smiley: